### PR TITLE
Revert "Revert "Show lockable stages in disabled mode for non-verified teachers""

### PR DESCRIPTION
**Pull request by dmcavoy**
Reverts code-dot-org/code-dot-org#40209 Re-implements #40068 

When reviewing the eyes diffs for the original change I noticed this one:

Screen Shot 2021-04-22 at 11 43 02 AM

Its a hidden lesson for a teacher and the lesson is getting faded out. We only want the fading for students. This PR re-implements all the changes from before while making sure we only fade out rows that are hidden for students. Example of what it looks like for the teacher now:

Screen Shot 2021-04-22 at 9 02 59 PM

### CODE REVIEW

File: apps/src/templates/progress/ProgressLesson.jsx [hunk 1]
```diff
@@ -206,7 +206,9 @@ class ProgressLesson extends React.Component {
         <div
           style={{
             ...styles.main,
-            ...((hiddenForStudents || isLockedForUser) && styles.translucent)
+            ...(((hiddenForStudents && !viewAs === ViewType.Teacher) ||
```
Comment (jamescodeorg) [inline apps/src/templates/progress/ProgressLesson.jsx]: For my education, is there a difference between checking `viewAs === ViewType.Student` and `viewAs !== ViewType.Teacher`?  Should we be using one or the other more consistently?

**Review by jamescodeorg — APPROVED** [review]
LGTM.

Is there a UI test that we can add/modify that would have caught this?

File: apps/src/templates/progress/ProgressLesson.jsx (hunk 1 above)
Comment (dmcavoy) [reply apps/src/templates/progress/ProgressLesson.jsx]: My thinking was about signed out users but now that you bring it up I think signed out users act as students so we could probably use viewAs === ViewType.Student. I will update.

**Comment by dmcavoy** [comment]
> LGTM.
> 
> Is there a UI test that we can add/modify that would have caught this?

I added a couple unit tests to check this going forward.

**Comment by davidsbailey** [comment]
> Is there a UI test that we can add/modify that would have caught this?

@Hamms and I had a long discussion about testing this morning. one thing we both agreed on is that it's best to cover any behavior with the smallest size test possible (where unit < integration < UI < eyes). in this case it seems possible to cover this behavior with a unit test, so that's what I'd recommend here. this has the advantage of running faster, being more reliable, and telling you more specifically what you broke (credit: Paul's [slides](https://docs.google.com/presentation/d/1RisVcJU3VwJxwV0FIUppGDOaBKvwABACzUfqyNP8Duk/edit)).

**Comment by dmcavoy** [comment]
> > Is there a UI test that we can add/modify that would have caught this?
> 
> @Hamms and I had a long discussion about testing this morning. one thing we both agreed on is that it's best to cover any behavior with the smallest size test possible (where unit < integration < UI < eyes). in this case it seems possible to cover this behavior with a unit test, so that's what I'd recommend here. this has the advantage of running faster, being more reliable, and telling you more specifically what you broke (credit: Paul's [slides](https://docs.google.com/presentation/d/1RisVcJU3VwJxwV0FIUppGDOaBKvwABACzUfqyNP8Duk/edit)).

Great! Thats what I ended up doing

**Pull request merged**